core/common/src/Clock.kt

```diff
@@ -53,5 +53,19 @@ private class InstantTimeMark(private val instant: Instant, private val clock: C
     override fun minus(duration: Duration): TimeMark = InstantTimeMark(instant - duration, clock)
 }
 
+/**
+ * Returns the [Clock] by storing an initial [TimeMark] using [TimeSource.markNow] and [returns][Clock.now] the elapsed
+ * time using [TimeMark.elapsedNow] plus the provided [offset].
+ *
+ * This clock stores the initial [TimeMark], so repeatedly creating [Clock]s from the same [TimeSource] results
```

Comment:
```suggestion
 * This clock stores the [TimeMark] at the moment of creation, so repeatedly creating [Clock]s from the same [TimeSource] results
```
I think this better conveys what happens.